webapp/src/components/dataset_input/LabeledImageDatasetInput/ByGroup.vue

```diff
@@ -67,7 +67,13 @@ function refreshWatcher() {
         ([label, files]) =>
           files.value?.map((f) => [label, f] as const)?.toArray() ?? [],
       );
-
+      // shuffle the filenames o.w. they are ordered by labels
+      for (let i = 0; i < expanded.length; i++) {
+        const j = Math.floor(Math.random() * i)
+        const swap = expanded[i]
+        expanded[i] = expanded[j]
+        expanded[j] = swap
+      }
```

Comment:
that shouldn't influence the training, it's for a nicer output in the prediction yes?
if that's the case, consider moving the shuffling there

Reply:
No it actually makes the training fail if the data stays ordered by labels. Batches are created sequentially, and because we don't implement any dataset-level shuffling (only batch-level), most of the batches contain a single class which makes the model updates too wild to learn anything. 
I implemented this file shuffling as a temporary placeholder until we have some kind of dataset shuffling